src/Language/PureScript/Bundle.hs

```diff
@@ -360,61 +388,158 @@ strValue str = go $ drop 1 str
   go (x : xs) = x : go xs
   go "" = ""
 
-commaList :: JSCommaList a -> [a]
-commaList JSLNil = []
-commaList (JSLOne x) = [x]
-commaList (JSLCons l _ x) = commaList l ++ [x]
-
 trailingCommaList :: JSCommaTrailingList a -> [a]
-trailingCommaList (JSCTLComma l _) = commaList l
-trailingCommaList (JSCTLNone l) = commaList l
+trailingCommaList (JSCTLComma l _) = fromCommaList l
+trailingCommaList (JSCTLNone l) = fromCommaList l
+
+identName :: JSIdent -> Maybe String
+identName (JSIdentName _ ident) = Just ident
+identName _ = Nothing
+
+exportStatementIdentifiers :: JSStatement -> [String]
+exportStatementIdentifiers (JSVariable _ jsExpressions _) =
+  varNames jsExpressions
+exportStatementIdentifiers (JSConstant _ jsExpressions _) =
+  varNames jsExpressions
+exportStatementIdentifiers (JSLet _ jsExpressions _) =
+  varNames jsExpressions
+exportStatementIdentifiers (JSClass _ jsIdent _ _ _ _ _) =
+  maybeToList . identName $ jsIdent
+exportStatementIdentifiers (JSFunction _ jsIdent _ _ _ _ _) =
+  maybeToList . identName $ jsIdent
+exportStatementIdentifiers (JSGenerator _ _ jsIdent _ _ _ _ _) =
+  maybeToList . identName $ jsIdent
+exportStatementIdentifiers _ = []
+
+varNames :: JSCommaList JSExpression -> [String]
+varNames = mapMaybe varName . fromCommaList
+  where
+  varName (JSVarInitExpression (JSIdentifier _ ident) _) = Just ident
+  varName _ = Nothing
+
+sp :: JSAnnot
+sp = JSAnnot tokenPosnEmpty [ WhiteSpace tokenPosnEmpty " " ]
+
+stringLiteral :: String -> JSExpression
+stringLiteral s = JSStringLiteral JSNoAnnot $ "\"" ++ s ++ "\""
```

Comment:
I think this is a bit dodgy - is language-javascript expecting the string here to be the same as it appears in the JS source code? Won't that cause problems if `s` contains characters like `\n` or `"`? I probably would have expected `language-javascript` to provide this function for us.

Reply:
Agreed, but I just renamed the existing `str` function 😬

Reply:
Oh haha, never mind then.